middleware/proxy/reverseproxy.go

```diff
@@ -147,14 +147,43 @@ func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request, extr
 	}
 	defer res.Body.Close()
 
-	for _, h := range hopHeaders {
-		res.Header.Del(h)
-	}
+	if res.StatusCode == http.StatusSwitchingProtocols && outreq.Header.Get("Upgrade") == "websocket" {
```

Comment:
Should `outreq.Header.Get("Upgrade")` be `res.Header.Get("Upgrade")` to be sure the backend server truly does support websockets at this endpoint?

This is my last question - then I'll merge in :smile: